state/protocol/badger/state.go

```diff
@@ -503,6 +511,26 @@ func (state *State) bootstrapSporkInfo(root protocol.Snapshot) func(*badger.Txn)
 	}
 }
 
+// indexFirstHeight indexes the first height for the epoch, as part of bootstrapping.
+// No errors are expected during normal operation.
+func indexFirstHeight(epoch protocol.Epoch) func(*badger.Txn) error {
```

Comment:
there are some benign sentinel errors that `epoch.FirstHeight()` could return: https://github.com/onflow/flow-go/blob/a3f746881bf1aa7ced07e9527c3532b3d81d3e15/state/protocol/badger/state.go#L522

Therefore, I feel the following statement makes a strong assumption about the usage of this function:
> // No errors are expected during normal operation. 

In my experience, avoiding assumptions about the caller makes code more general and easier to maintain. Why don't we document the error cases here and let the caller decide whether these are expected (?)

Reply:
This function is private to its package, single-purpose, and limited in scope to bootstrapping. It is in the same package and file as the broader bootstrapping logic, and is called from one other function. Personally I think for a function like this it is more appropriate for it to make (and document) strong requirements on its inputs and enforce them. (This function was missing the documentation about requirements for the input - I've added those now). 

Without placing any expectations on the input value, it's true that we might observe benign sentinels here. However, in the context of this function's use, we explicitly don't expect those possible sentinels. Therefore I think it makes sense to "filter" those possible sentinels at this point in the stack, by treating them as an exception.

Reply:
thanks for the explanation. agree